lib/RGBot.js

```diff
@@ -400,39 +769,142 @@ const RGBot = class {
      *
      * The Bot will avoid the given Entity.
      * @param {Entity} entity The Entity to avoid
-     * @param {object} options={} Optional parameters
+     * @param {object} [options={}] Optional parameters
      * @param {number} [options.reach=5] The Bot will not move within this reach of the Entity
      * @returns {Promise<void>}
      */
     async avoidEntity(entity, options = {}) {
         const reach = options.reach || 5;
         if (!entity) {
-            console.error(`avoidEntity: Entity was null or undefined`);
+            this.#logWarn(`avoidEntity: Entity was null or undefined`);
         } else {
-            this.#log(`Avoiding ${this.getEntityName(entity)} at a minumum reach of ${reach}`);
+            this.#logDebug(`Avoiding ${this.getEntityName(entity)} at a minimum reach of ${reach}`);
             this.#bot.pathfinder.setGoal(new GoalInvert(new GoalFollow(entity, reach)), true);
         }
     }
 
     /**
      * <i><b>Experimental</b></i>
      *
-     * The Bot will attack the given Entity one time.
-     * This <i>will not</i> move the Bot towards its target: calling {@link `followEntity`} is recommended for staying within attack range of the target as it moves.
-     * @param {Entity} entity
-     * @returns {void}
+     * This will move the bot to within range of the target, equip the most powerful weapon in the bot inventory,
+     * and attack the target 1 time.  To finish off a target, this method must be called until the target is dead.
+     *
+     * @param {Entity} entity - The entity to attack
+     * @param {object} [options={}] Optional parameters
+     * @param {number} [options.reach=2] How close to get to the target before attacking
+     * @param {Item} [options.attackItem=undefined] An item in the bot inventory to use for the attack
+     * @returns {boolean} - did we successfully attack
+     *
+     * @example
+     * let target = //<someEntity>
+     * while (target.isValid) {
+     *     attackEntity(target)
+     * }
+     *
+     * Note: This method currently only uses melee weapons
      */
-    attackEntity(entity) {
-        if (!entity) {
-            console.error(`attackEntity: Entity was null or undefined`);
-        } else {
+    async attackEntity(entity, options={}) {
+        const reach = options.reach || 2;
+        if (entity) {
             try {
-                this.#log(`Attacking ${this.getEntityName(entity)}`);
-                this.#bot.attack(entity, true);
-            } catch (err) {
-                console.error(`Error attacking target: ${this.getEntityName(entity)}`, err)
+                this.#logDebug(`Following and attacking entity ${entity.username || entity.name}`)
+                if (entity.isValid) {
```

Comment:
Remove `if` and instead right before have something like
```
if (!entity.isValid) {
    this.#logWarn(`Could not follow and attack entity $entity because it is not a valid entity (as defined by entity.isValid)`)
}
...
// code that was in if statement before
```